plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,434 @@
+  # timeout = "5s"
+
+  ## List of statistics which are collected
+  # At least one is required
+  # Allowed values: server, databases, indexes, collections
+  #
+  # stats_include = ["server", "databases", "indexes", "collections"]
+
+  ## List of db where database stats are collected
+  ## If empty, all db are concerned
+  # db_stats_dbs = []
+
+  ## List of db where index status are collected
+  ## If empty, all indexes from all db are concerned
+  # index_stats_dbs = []
+
+  ## List of db where collection status are collected
+  ## If empty, all collections from all db are concerned
+  # collection_stats_dbs = []
+`
+
+// SampleConfig ...
+func (r *RavenDB) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description ...
+func (r *RavenDB) Description() string {
+	return "Reads metrics from RavenDB servers via the Monitoring Endpoints"
+}
+
+// Gather ...
+func (r *RavenDB) Gather(acc telegraf.Accumulator) error {
+	err := r.ensureClient()
+	if nil != err {
+		r.Log.Errorf("Error with Client %s", err)
+		return err
+	}
+
+	var wg sync.WaitGroup
+
+	for _, statToCollect := range r.StatsInclude {
+		switch statToCollect {
+		case "server":
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				r.gatherServer(acc)
+			}()
+		case "databases":
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				r.gatherDatabases(acc)
+			}()
+		case "indexes":
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				r.gatherDatabases(acc)
+			}()
+		case "collections":
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				r.gatherCollections(acc)
+			}()
+		}
+	}
+
+	wg.Wait()
+
+	return nil
+}
+
+func (r *RavenDB) ensureClient() error {
+	if r.client != nil {
+		return nil
+	}
+
+	tlsCfg, err := r.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+	tr := &http.Transport{
+		ResponseHeaderTimeout: r.Timeout.Duration,
+		TLSClientConfig:       tlsCfg,
+	}
+	r.client = &http.Client{
+		Transport: tr,
+		Timeout:   r.Timeout.Duration,
+	}
+
+	return nil
+}
+
+func (r *RavenDB) requestJSON(u string, target interface{}) error {
+	req, err := http.NewRequest("GET", u, nil)
+	if err != nil {
+		return err
+	}
+
+	resp, err := r.client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	r.Log.Debugf("%s: %s", u, resp.Status)
+	if resp.StatusCode >= 400 {
+		return fmt.Errorf("invalid response code to request '%s': %d - %s", r.URL, resp.StatusCode, resp.Status)
+	}
+
+	return json.NewDecoder(resp.Body).Decode(target)
+}
+
+func (r *RavenDB) gatherServer(acc telegraf.Accumulator) {
+	serverResponse := &serverMetricsResponse{}
+
+	err := r.requestJSON(r.requestUrlServer, &serverResponse)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+
+	tags := map[string]string{
+		"cluster_id": serverResponse.Cluster.Id,
+		"node_tag":   serverResponse.Cluster.NodeTag,
+		"url":        r.URL,
+	}
+
+	if serverResponse.Config.PublicServerUrl != nil {
+		tags["public_server_url"] = *serverResponse.Config.PublicServerUrl
```

Comment:
Should you check for the string being blank here as well? if that's at all possible to occur

Reply:
no need to check - if option is defined in RavenDB it can't be empty.